lib/action_controller/serialization.rb

```diff
@@ -30,7 +30,7 @@ def get_serializer(resource, options = {})
         serializable_resource.serialization_scope ||= serialization_scope
         serializable_resource.serialization_scope_name = _serialization_scope
         begin
-          serializable_resource.adapter
+          serializable_resource.tap(&:adapter)
```

Comment:
Yeah, we need to do a better job with the class's api... Should have less branching in controlker

Needs to return self to be ibstrumented at this time

B mobile phone

> On Nov 8, 2015, at 4:16 PM, Lucas Hosseini <notifications@github.com> wrote:
> 
> In lib/action_controller/serialization.rb:
> 
> > @@ -30,7 +30,7 @@ def get_serializer(resource, options = {})
> >          serializable_resource.serialization_scope ||= serialization_scope
> >          serializable_resource.serialization_scope_name = _serialization_scope
> >          begin
> > -          serializable_resource.adapter
> > +          serializable_resource.tap(&:adapter)
> Could you explain the rationale here?
> serializable_resource.tap(&:adapter) returns serializable_resource but calls the serializable_resource.adapter method in the meantime. Is this to ensure the serializers get built and the CollectionSerializer::NoSerializerError gets raised if necessary?
> 
> —
> Reply to this email directly or view it on GitHub.
>